cli/server/server.go

```diff
@@ -588,6 +588,17 @@ Main:
 				if p2pNotary != nil && serv.IsInSync() {
 					p2pNotary.Start()
 				}
+				srMod.SetUpdateValidatorsCallback(nil)
+				sr.Shutdown()
+				sr, err = stateroot.New(cfgnew.ApplicationConfiguration.StateRoot, srMod, log, chain, serv.BroadcastExtensible)
+				if err != nil {
+					log.Error("failed to create state validation service", zap.Error(err))
+					break // The show must go on.
+				}
+				if serv.IsInSync() {
+					sr.Start()
+				}
+				serv.AddExtensibleService(sr, stateroot.Category, sr.OnPayload)
```

Comment:
Technically, race condition is possible here: if server is not in sync, `sr.Start()` won't be executed. After that if server comes to the synced state before `serv.AddExtensibleService(sr, stateroot.Category, sr.OnPayload)` execution, then stateroot service will never be started, as it's not yet in the list of services.

Other services don't have such problem, because service registration happens before the `serv.IsInSync()` check.